test/services/samples/destroy_service_test.rb

```diff
@@ -126,5 +126,60 @@ def setup
       assert_equal({}, subgroup12b.reload.metadata_summary)
       assert_equal({ 'metadatafield1' => 1, 'metadatafield2' => 1 }, group12.reload.metadata_summary)
     end
+
+    test 'samples count updated after single sample deletion' do
+      # Reference group/projects descendants tree:
+      # group12 < subgroup12b (project30 > sample 33)
+      #    |
+      #    ---- < subgroup12a (project29 > sample 32) < subgroup12aa (project31 > sample34 + 35)
+      group12 = groups(:group_twelve)
+      subgroup12a = groups(:subgroup_twelve_a)
+      subgroup12b = groups(:subgroup_twelve_b)
+      subgroup12aa = groups(:subgroup_twelve_a_a)
+      project31 = projects(:project31)
+      sample34 = samples(:sample34)
```

Comment:
Could you refactor these into this test file's setup and remove the individual setups as many tests use these fixtures. Feel free to look through the other test files you worked in and make the same change if they're called many times as well.